gui/src/components/TokenBalance.js

```diff
@@ -13,40 +15,46 @@ const formatBalance = (val) => {
 }
 
 const valAndTooltip = ({value, title, error}) => {
+  let val, tooltip
+
   if (error) {
-    return {
-      val: error,
-      tooltip: 'Error'
-    }
+    val = error
+    tooltip = 'Error'
   } else if (value == null) {
-    return {
-      val: '...',
-      tooltip: 'Loading...'
-    }
+    val = '...'
+    tooltip = 'Loading...'
+  } else {
+    const balance = formatBalance(value)
+    val = balance.formatted
+    tooltip = balance.unformatted
   }
 
-  return {
-    val: formatBalance(value).formatted,
-    tooltip: formatBalance(value).unformatted
-  }
+  return {val, tooltip}
 }
 
 const TokenBalance = props => {
   const {val, tooltip} = valAndTooltip(props)
 
   return (
-    <MetaInfo
-      className={props.className}
-      title={props.title}
-      value={val}
-      tooltip={tooltip}
-    />
+    <PaddedCard>
+      <Typography gutterBottom variant='headline' component='h2'>
+        {props.title}
+      </Typography>
+      <Typography variant='body1' color='textSecondary'>
+        <Tooltip title={tooltip} placement='left'>
+          <span>{val}</span>
+        </Tooltip>
+      </Typography>
+    </PaddedCard>
   )
 }
 
 TokenBalance.propTypes = {
   title: PropTypes.string.isRequired,
-  value: PropTypes.string,
+  value: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number
```

Comment:
Let me know if I'm being pedantic here, but the idea of a variable being multiple types seems like a smell to me. Can we coerce this to always be a string?

Reply:
@dimroc it can always be a string now